clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -67,7 +68,8 @@ public FutureResponse(AbstractResponse responseBody, boolean disconnected, Reque
     private final Set<String> ready = new HashSet<>();
     private final Map<Node, Long> blackedOut = new HashMap<>();
     private final Queue<ClientRequest> requests = new ArrayDeque<>();
-    private final Queue<ClientResponse> responses = new ArrayDeque<>();
+    // Use concurrent queue for responses so that responses may be updated during poll() from a different thread.
+    private final Queue<ClientResponse> responses = new ConcurrentLinkedDeque<>();
```

Comment:
@ijuma I thought of making all of them `ConcurrentLinkedDeque` since there is no harm in doing so. But the close test requires only `responses` to be thread-safe. I wasn't sure if it is confusing to use concurrent queues when there is no expectation of thread-safety. What do you think?